rln/src/pm_tree_adapter.rs

```diff
@@ -60,6 +60,89 @@ fn get_tmp() -> bool {
 
 pub struct PmtreeConfig(Config);
 
+pub struct PmtreeConfigBuilder {
+    path: Option<PathBuf>,
+    temporary: Option<bool>,
+    cache_capacity: Option<u64>,
+    flush_every_ms: Option<u64>,
+    mode: Option<Mode>,
+    use_compression: Option<bool>,
+}
+
+impl PmtreeConfigBuilder {
```

Comment:
Cannot you use a crate to create automatically a builder. Maybe using crate: bon or crate: derive_builder ?